### PR TITLE
bpo-38641: Add support of starred expressions in return/yield to lib2to3

### PR DESCRIPTION
**Pull request by vemel**
### Goal

[bpo-32117](https://bugs.python.org/issue32117) is fixed, but unfortunately `lib2to3` still does not support unpacking in `return` and `yield` without parentheses.

```python
def test():
    my_list = ["value2", "value3"]
    yield "value1", *my_list
    return "value1", *my_list
```

### Changes

- Use `testlist_star_expr` in `return_stmt`
- Use `testlist_star_expr` in `yield_arg`

### Summary

This is a backwards compatible change, because `testlist_star_expr` supports `test` as well like `testlist` does.

https://bugs.python.org/issue38641

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@vemel

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by pablogsal** [comment]
Hi @vemel and thanks for the change!

Could you sign the CLA and add a NEWS entry? Thanks!

File: Misc/NEWS.d/next/Library/2019-10-30-15-31-09.bpo-38641.HrTL9k.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Added starred expressions support to ``return`` and ``yield`` statements.
```
Comment (isidentical) [inline Misc/NEWS.d/next/Library/2019-10-30-15-31-09.bpo-38641.HrTL9k.rst]: Can you specify this is a bug related to `lib2to3` here, also you can add a `Patch by <name>.` attribution at the end of news.

File: Misc/ACKS [hunk 2]
```diff
@@ -1896,3 +1896,4 @@ Robert Leenders
 Tim Hopper
 Dan Lidral-Porter
 Ngalim Siregar
+Vlad Emelianov
```
Comment (isidentical) [inline Misc/ACKS:1913]: Please resolve merge conflict here

File: Lib/lib2to3/tests/data/py3_test_grammar.py [hunk 3]
```diff
@@ -473,14 +473,25 @@ def test_inner(extra_burning_oil = 1, count=0):
...
         x = g2()
+        x3 = g3()
         check_syntax_error(self, "class foo:return 1")
 
     def testYield(self):
+        # 'yield' [yield_arg]
+        def g1(): yield 1
+        yield_list = [2, 3]
+        def g2(): yield 1, *yield_list
+        def g3(): yield from iter(yield_list)
```
Comment (isidentical) [inline Lib/lib2to3/tests/data/py3_test_grammar.py:491]: Maybe also a syntax error test for `check_syntax_error(self, "def g4(): yield from *a")`

File: Misc/ACKS (hunk 2 above)
Comment (vemel) [reply Misc/ACKS:1913]: Will do

Comment (vemel) [reply Misc/ACKS:1913]: Resolved.

File: Misc/NEWS.d/next/Library/2019-10-30-15-31-09.bpo-38641.HrTL9k.rst (hunk 1 above)
Comment (vemel) [reply Misc/NEWS.d/next/Library/2019-10-30-15-31-09.bpo-38641.HrTL9k.rst]: Updated, thank you!

File: Lib/lib2to3/tests/data/py3_test_grammar.py (hunk 3 above)
Comment (vemel) [reply Lib/lib2to3/tests/data/py3_test_grammar.py:491]: Added, good idea!

**Comment by vemel** [comment]
Looks like unit test for MacOS are broken, but it is not related to current PR. Probably it makes sense to fix them in another PR.

**Review by isidentical — APPROVED** [review]
LGTM except for the ACKS file, I think you did something wrong while you were rebasing/merging. Can you check it again @vemel

File: Misc/ACKS (hunk 2 above)
Comment (isidentical) [reply Misc/ACKS:1913]: Doesn't look resolved, can you please check this again.

**Comment by vemel** [comment]
> ACKS

Indeed. Fixed.

**Review by isidentical — APPROVED** [review]

**Review by pablogsal — APPROVED** [review]
LGTM

Thanks, @vemel for the PR and @isidentical for the review :)

**Comment by isidentical** [comment]
Thanks @vemel for the PR!

**Pull request merged**